City-of-Helsinki/drupal-helfi-platform-config
# Uhf 4464 poc

# Events-paragraph [UHF-4464](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4464)
This branch contains code for both the subtasks of the issue.

## What was done
- Add new paragraph type _events_.  Make it addable to content fields on landing page and basic page types.
- Add classes to retrieve data from LinkedEvents API based on the given tapahtumat.hel.fi url
- Add basic JS functionality to query events based on the url and form a simple list from them

## How to install
- Pick a Drupal with `helfi_platform_config` installed.  Run `composer reinstall helfi_platform_config --prefer-source`.
- Cd into the module directory. Checkout to branch. `make shell` into the container and run `drush en helfi_events -y`. 

## How to test
- Add a new landing page / basic page or edit an existing one. You should be able to add a paragraph of type events to to `field_content` (landing page and basic page) or to `field_lower_content` (basic page only).
- Add a new events-paragraph. Get the url for `field_api_url` by searching something from https://tapahtumat.hel.fi/fi/home and copying the url from the address bar.
- Viewing the page, you should see a matching list of events on the page that you saw at tapahtumat.hel.fi
